sources/logdog/src/log_request.rs

```diff
@@ -205,12 +218,114 @@ where
     Ok(())
 }
 
+/// Copies all file matching the glob pattern given by `request.instructions` to the tempdir with filename and path
+/// same as source file.
+fn handle_glob_request<P>(request: &LogRequest<'_>, tempdir: P) -> Result<()>
+where
+    P: AsRef<Path>,
+{
+    let glob_paths = glob(request.instructions).context(error::ParseGlobPattern {
+        pattern: request.instructions.to_string(),
+    })?;
+    for entry in glob_paths {
+        if let Ok(path) = entry {
+            if path.is_dir() {
+                // iterate the directory and copy files in the directory, nested
+                // sub-directories files will be copied only if matched by glob pattern.
+                for candidate in WalkDir::new(&path)
+                    .max_depth(1)
+                    .into_iter()
+                    .flat_map(|e| e.context(error::DirectoryWalk))
+                    .map(|e| e.into_path())
```

Comment:
```suggestion
                    .into_iter()
                    .filter_map(|e| e.ok())
```
For the glob piece, we do `if let Ok` to ignore errors; I think it'd be fine to have the same behavior here.  (The suggestion is the standard way to do that with walkdir.)  We're trying to get as much information as possible, and we wouldn't want to stop collecting more files because of issues with a prior one.